fuel-vm/src/interpreter/initialization.rs

```diff
@@ -41,6 +45,33 @@ where
     ) -> Result<(), RuntimeError<S::DataError>> {
         tx.prepare_init_execute();
         self.tx = tx;
+        self.input_contracts = self
```

Comment:
So we might be taking a bit of a hit on performance because of this, but at least its constant time and not an attack vector?

Reply:
Yes, we calculate it each time now during initialization and later all related operations will be fast